java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -6338,6 +6338,56 @@ void testBoundsCheckedGather() {
     }
   }
 
+
+  @Test
+  void testScatterTable() {
+    try (Table srcTable = new Table.TestBuilder()
+            .column(1, 2, 3, 4, 5)
+            .column("A", "AA", "AAA", "AAAA", "AAAAA")
+            .decimal32Column(-3, 1, 2, 3, 4, 5)
+            .decimal64Column(-8, 100001L, 200002L, 300003L, 400004L, 500005L)
+            .build();
+         ColumnVector scatterMap = ColumnVector.fromInts(0, 2, 4, -2);
+         Table targetTable = new Table.TestBuilder()
+            .column(-1, -2, -3, -4, -5)
+            .column("B", "BB", "BBB", "BBBB", "BBBBB")
+            .decimal32Column(-3, -1, -2, -3, -4, -5)
+            .decimal64Column(-8, -100001L, -200002L, -300003L, -400004L, -500005L)
+            .build();
+         Table expected = new Table.TestBuilder()
+            .column(1, -2, 2, 4, 3)
+            .column("A", "BB", "AA", "AAAA", "AAA")
+            .decimal32Column(-3, 1, -2, 2, 4, 3)
+            .decimal64Column(-8, 100001L, -200002L, 200002L, 400004L, 300003L)
+            .build();
+         Table result = srcTable.scatter(scatterMap, targetTable, false)) {
+      assertTablesAreEqual(expected, result);
+    }
+  }
+
+  @Test
+  void testScatterScalars() {
+    Scalar[] srcScalars = new Scalar[] { Scalar.fromInt(0), Scalar.fromString("A") };
```

Comment:
This can leak the first scalar if we fail to create the second scalar.  I'd leverage the try-with-resources already being used below via something like this which removes the need for the `finally` block.
```java
  try(Scalar s1 = Scalar.fromInt(0);
      Scalar s2 = Scalar.fromString("A");
      ColumnVector scatterMap = ....
      ...
      Table result = Table.scatter(new Scalar[] { s1, s2 }, scatterMap, targetTable, false)) {
```